UsingTheServer.md

```diff
@@ -0,0 +1,130 @@
+| `public Response` | `createResponse(Request request)` |
+
+See [Example Controller](#example-controller) for more information.
+
+**Response**
+
+| Type            | Method                                 |
+| --------------- | -------------------------------------- |
+| `constructor`   | `Response(Header header, byte[] body)` |
+| `public Header` | `getHeader()`                          |
+| `public byte[]` | `getBody()`                            |
+
+**ResponseBuilder**
+
+| Type                     | Method                                   |
+| ------------------------ | ---------------------------------------- |
+| `public ResponseBuilder` | `addBody(String body)`                   |
+|                          | `addBody(byte[] body)` <br><br>Add body to response.  Overloaded string variation will convert body to a byte array. |
+| `public ResponseBuilder` | `addStatus(int code)` <br><br> Adds HTTP status code to header response line. |
+| `public ResponseBuilder` | `addHeader(String title, String content` <br><br> Adds individual headers to Header instance. For example, `addHeader(Content-Type, text/html)` adds Content-Type: text/html to the header. |
+| `public Response`        | `build()` <br><br>Creates a `Response` instance with information added via other `ResponseBuilder` methods. |
+
+**Header**
+
+| Type | Method |
+| ---- | ------ |
+| `public void` | `addStatus(int code)` <br><br>Adds HTTP status code to header response line. |
+| `public void` | `addHeader(String title, String content)` <br><br>Adds individual headers to Header instance. For example, `addHeader(Content-Type, text/html)` adds Content-Type: text/html to the header. |
+| `public String` | `formatToString()` <br><br>Returns the full header formatted for HTTP. |
+
+**Interface ErrorController**
+
+The `ErrorController` interface extends the `Controller` interface.  Implementations should implement `Controller` and `ErrorController` methods.
+
+| Type                     | Method                     |
+| ------------------------ | -------------------------- |
+| `public ErrorController` | `updateStatus(int status)` <br><br>The server uses this method to update status code based on the type of client error. |
+
+**Interface Logger**
+
+| Type          | Method               |
+| ------------- | -------------------- |
+| `public void` | `log(String status)` |
+
+### Example Controller
+
+An example `Controller` from the cobspec application is provided below:
+
+```
+public class RedirectController implements Controller {
+    public Response createResponse(Request request) {
+        return new ResponseBuilder()
+                    .addStatus(302)
+                    .addHeader("Location", "/")
+                    .build();
+    }
+}
+```
+
+Use of the `ResponseBuilder` is not required but is provided for convenience.
+
+Your `Controller` may contain whatever additional methods are needed to create contents of your response.
```

Comment:
The public API is well documented, but I still don't understand how to set up and use the server in my own project?, maybe a simple example of wiring it all up would help?

Reply:
+1